tests/unit/testplan/testing/multitest/test_result.py

```diff
@@ -91,6 +100,33 @@ def test_group_marking_multitest(mockplan, flag):
     for desc, line_no in expected.items():
         assert assertions[desc]["line_no"] == line_no
 
+    test = MultiTest(
+        name="ParametrizedGroupMarking",
+        suites=[ParametrizedGroupMarking()],
+        testcase_report_target=flag,
+    )
+    test.cfg.parent = mockplan.cfg
+    test.run()
+    assertions = {
+        entry["description"]: entry
+        for entry in test.report.flatten()
+        if isinstance(entry, dict) and entry["meta_type"] == "assertion"
+    }
+    expected = {
+        "A0": get_line_no(ParametrizedGroupMarking.case, 2),
+        "B0": get_line_no(ParametrizedGroupMarking.case, 3)
+        if flag
+        else get_line_no(helper, 1),
+        "C0": get_line_no(intermediary, 2),
+        "A1": get_line_no(ParametrizedGroupMarking.case, 2),
+        "B1": get_line_no(ParametrizedGroupMarking.case, 3)
+        if flag
+        else get_line_no(helper, 1),
+        "C1": get_line_no(intermediary, 2),
```

Comment:
No concern, but maybe at this point this became a bit hard to read due to multiple ternary-like, at least for me. Why we have two times the `if flag else` piece?

Reply:
ugly code, I have updated it.